core/src/main/java/org/apache/carbondata/core/metadata/datatype/DecimalConverterFactory.java

```diff
@@ -348,6 +368,32 @@ public void fillVector(Object valuesToBeConverted, int size,
             vector.putDecimal(i, value, precision);
           }
         }
+      } else if (valuesToBeConverted instanceof byte[]) {
```

Comment:
Shall we avoid this code and reuse ? Same as above.

Reply:
yes, check and resue the code for this also

Reply:
done